src/SosThreadingTools/SosThreadingTools.targets

```diff
@@ -0,0 +1,45 @@
+<Project>
+  <UsingTask TaskName="ReplaceFileText" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll">
+    <ParameterGroup>
+      <InputFilename ParameterType="System.String" Required="true" />
+      <OutputFilename ParameterType="System.String" Required="true" />
+      <MatchExpression ParameterType="System.String" Required="true" />
+      <ReplacementText ParameterType="System.String" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include="System.Core" />
+      <Using Namespace="System" />
+      <Using Namespace="System.IO" />
+      <Using Namespace="System.Text.RegularExpressions" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+            File.WriteAllText(
+                OutputFilename,
+                Regex.Replace(File.ReadAllText(InputFilename), MatchExpression, ReplacementText)
+                );
+          ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <PropertyGroup>
+    <TargetsForTfmSpecificContentInPackage>$(TargetsForTfmSpecificContentInPackage);PackBuildOutputs</TargetsForTfmSpecificContentInPackage>
+    <GenerateNuspecDependsOn>StampAndIncludeGalleryManifest;$(GenerateNuspecDependsOn)</GenerateNuspecDependsOn>
+  </PropertyGroup>
+
+  <Target Name="PackBuildOutputs" DependsOnTargets="DebugSymbolsProjectOutputGroup">
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="$(TargetPath);@(DebugSymbolsProjectOutputGroupOutput)" PackagePath="tools\$(RuntimeIdentifier)\" />
+    </ItemGroup>
+  </Target>
+  <Target Name="StampAndIncludeGalleryManifest" DependsOnTargets="GetBuildVersion" BeforeTargets="_GetPackageFiles">
+    <ReplaceFileText
+      InputFilename="GalleryManifest.xml"
+      OutputFilename="$(IntermediateOutputPath)GalleryManifest.xml"
+      MatchExpression="\$Version\$"
+      ReplacementText="$(BuildVersion)" />
```

Comment:
😨  

💭 Is there any way we can leverage the functionality provided by the VSSDK for rewriting the source.extension.vsixmanifest files?

Reply:
VS SDK's support is very specifically done from their vsixmanifest parser. I can't imagine how that would be applicable here. Nor do I want a dependency on the VS SDK inside our repo since that takes us further from `dotnet build` support.